java-lib/src/main/java/com/wavefront/ingester/GraphiteDecoder.java

```diff
@@ -26,19 +27,24 @@
       .appendOptionalTimestamp().whiteSpace()
       .appendAnnotationsConsumer().whiteSpace().build();
   private final String hostName;
+  private LinkedHashSet<String> customSourceTags;
 
-  public GraphiteDecoder() {
+  public GraphiteDecoder(LinkedHashSet<String> customSourceTags) {
```

Comment:
Alright, I deliberately used the LinkedHashSet here to ensure the ordering would always work as expected, because I think having it ordered is the only use case that ever makes sense here. I wanted to make sure that even if someone was changing what the PushAgent passes down the road they wouldn't wonder "Why LinkedHashSet, I'll just use HashSet" and accidentally break it.

If you're sure I can certainly change it though. Do you want it changed to Set in all arg lists, like in IngesterFormatter.drive() as well ?